contracts/bprotocol/Registry.sol

```diff
@@ -97,6 +97,15 @@ contract Registry is Ownable {
         emit RevokeDelegate(msg.sender, _avatar, delegatee);
     }
 
+    function delegateAndExecuteOnce(address delegatee, address payable target, bytes calldata data) payable external {
+        // make sure there is an avatar
+        getAvatar(msg.sender);
+        delegateAvatar(delegatee);
+        (bool succ, bytes memory err) = target.call.value(msg.value)(data);
```

Comment:
Was just thinking:
- An attacker can create an Avatar for Registry contract. I don't think this will harm the contract.